kornia/contrib/connected_components.py

```diff
@@ -51,8 +52,18 @@ def connected_components(image: Tensor, num_iterations: int = 100) -> Tensor:
     out = torch.arange(B * H * W, device=image.device, dtype=image.dtype).view((-1, 1, H, W))
     out[~mask] = 0
 
-    for _ in range(num_iterations):
-        out = F.max_pool2d(out, kernel_size=3, stride=1, padding=1)
-        out = torch.mul(out, mask)  # mask using element-wise multiplication
+    i = 0
+    while True:
+        i += 1
+        out_ = F.max_pool2d(out, kernel_size=3, stride=1, padding=1)
+        # mask using element-wise multiplication
+        out_ = torch.mul(out_, mask)
+        # stop if converged
+        if torch.all(out == out_):
```

Comment:
this check here  will fail due to numerical accuracy for floating point tensors. I suggest adding another metric (rmse or mse?) subject to a threshold. Worth to verify in the original connected components algorithm how is it done. /cc @ducha-aiki

Reply:
I changed it to `torch.allclose`.